libsignal-service/src/account_manager.rs

```diff
@@ -78,11 +75,14 @@ impl<Service: PushService> AccountManager<Service> {
     /// Equivalent to Java's RefreshPreKeysJob
     ///
     /// Returns the next pre-key offset and next signed pre-key offset as a tuple.
-    pub async fn update_pre_key_bundle(
+    pub async fn update_pre_key_bundle<R: rand::Rng + rand::CryptoRng>(
         &mut self,
-        store_context: StoreContext,
+        identity_store: &dyn IdentityKeyStore,
+        pre_key_store: &mut dyn PreKeyStore,
+        signed_pre_key_store: &mut dyn SignedPreKeyStore,
```

Comment:
Let's make an issue about Rustyfing these. Or do we want a `-service`-exposed `StoreContext` that encompasses those three?

FWIW, `StoreContext` basically was a `Box<dyn T>` implementation too :'-)

Reply:
Yes, the problem is just what happens to the type when you want to require `Clone` and push the responsibility of handling concurrent mutability to the implementer. If you wrap those in a `StoreContext`, you can't easily derive `Clone` (at least, I tried, and... 😪)

Reply:
Let's track this at #92